### DISCUSSION TITLE
Modify 0-on-boot.sh to consider pihole

### DISCUSSION THREAD

**Discussion by jetuser** [unifi-utilities/unifi-common#637] [General]
If you're running pihole on a UDM, oftentimes if the UDM updates the core os the cached apt packages you have locally will not install correctly (due to the OS updating) requiring you to go back and update the repository to get new packages to re-install systemd container to get machinectl back. 

Now obivously if you've updated your dns config to point to your pihole and systemd-container is down the automated reinstall won't work and you'll black hole your network. Can someone suggest a modification to the boot script to temporarily flip the dns back to a public dns to update/re-cache the necessary packages before re-enabling the pihole container and then re-pointing back to it?

As it stands, I have to manually flip the dns back to public and manually update/re-install systemd-container before i can get my pihole back up and running.

**Comment by m4n63** [comment]
What I do is that I just reinstall the udm-boot package everytime I upgrade, and that usually then runs all scripts for me and then I am up and running again. My 0-setup-system.sh looks like this:
```
#!/bin/bash
# This script installs systemd-container if it's not installed.
# Also links any containers from /data/custom/machines to /var/lib/machines.

if ! dpkg -l systemd-container | grep ii >/dev/null; then
    if ! apt -y install systemd-container debootstrap; then
        yes | dpkg -i /data/custom/dpkg/*.deb
    fi
fi

mkdir -p /var/lib/machines
for machine in $(ls /data/custom/machines/); do
        if [ ! -e "/var/lib/machines/$machine" ]; then
                ln -s "/data/custom/machines/$machine" "/var/lib/machines/"
                machinectl enable $machine
                machinectl start $machine
        fi
done
```

**Comment by jetuser** [comment]
The problem is less re-installing udm-boot. That persists fine across updates, the problem is the unifi updates often update the os so the cached packages cannot be used to reinstall systemd-container. I cannot run apt commands while the dns is redirected through my pihole after update since it black-holes all traffic. 

What I am suggesting is seeing if there's a good way to detect the OS has been updated, disable the pihole dns config and then pull the packages before re-enabling pihole configuration. 

Or otherwise have the system always bypass the pihole separately from the rest of the network
